packages/debugger-extension/src/index.ts

```diff
@@ -824,6 +825,16 @@ const sidebar: JupyterFrontEndPlugin<IDebugger.ISidebar> = {
         if (kernel && filters[kernel]) {
           sidebar.variables.filter = new Set<string>(filters[kernel]);
         }
+
+        const hide = setting.get('hideNativeFilteredKernelSources')
+          .composite as boolean;
+
+        if (hide) {
+          service.model.kernelSources.hiddenSources = filtersData;
```

Comment:
Maybe this is implementation details, sorry if it is the case :smile: 

IIUC, the list of "native kernel sources" is static: currently it comes from a JSON file, and maybe after a JEP, it will come from the debugger service (but will not change at runtime AFAIU). When it will come from the debugger service, we'll have to set a flag in the model to hide them or not.
IMO, we should only set a flag to the model here (e.g. `hideNativeKernelSource`), and set the list of native kernel source in the constructor of the model.

Reply:
I think this was not updated yet, right?